docs/book/v3/application-integration/usage-in-a-laminas-mvc-application.md

```diff
@@ -67,48 +67,42 @@ class AlbumForm extends Form implements InputFilterProviderInterface
 }
 ```
 
-## Using Form
+## Create Controller
 
-### Create Controller
-
-[Create a controller class](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-controller)
-and inject the form via the constructor, e.g.
-`module/Album/Controller/AlbumController.php`:
+[Create a controller class](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-controller) and inject the form element manager via the constructor, e.g. `module/Album/Controller/AlbumController.php`:
 
 ```php
 namespace Album\Controller;
 
 use Album\Form\AlbumForm;
-use Laminas\Form\FormInterface;
+use Laminas\Form\FormElementManager;
 use Laminas\Mvc\Controller\AbstractActionController;
 
 class AlbumController extends AbstractActionController
 {
-    /** @var FormInterface */
-    private $form;
-    
-    public function __construct(AlbumForm $form)
-    {
-        $this->form = $form;
-    }
+    public function __construct(
+        public readonly FormElementManager $formElementManager
+    ) {}
```

Comment:
I wonder if instead we could do the following

```php
    private AlbumForm $form;

    public function __construct(FormElementManager $formElementManager)
    {
        $this->form = $formElementManager->get(AlbumForm::class);
    }
```

This would ensure that we have a form of the expected type, and it would provide a slight performance benefit if the controller were re-used (e.g., if using something like OpenSwoole), as we wouldn't be pulling the form from the form element manager each time.

Reply:
@weierophinney maybe I misunderstand, but... forms are just dripping with state. OK, this is documenting MVC-land, but I would still encourage people to get a fresh one every time they need it, if just to encourage right thinking.

I like the idea of injecting `FormElementManager` though. Maybe a `private function getForm(): AlbumForm` could ensure the type?

Reply:
I agree with @kynx : a Form i neither a `Service` nor a `Value Object`. A Form is an `Entity`, stateful and mutable.
In a well structured application this shouldn't matter much as a Controller should be unique and single-purpose, but still the idea of having it as a constructor dependency feels wrong to me.
In my point of view, the perfect Controller have only `Service` dependencies, injected in the constructor and saved into `readonly` attributes; `Entities` have to be created where and when needed.

Reply:
> `Entities` have to be created where and when needed.

What is your concrete proposal? `$form = new AlbumForm();`?

Reply:
```php
final class MyController implements RequestHandlerInterface
{
    public function __construct(
        private readonly FormElementManager $formElementManager,
    ) {
    }

    public function handle(ServerRequestInterface $request): ResponseInterface
    {
        // Some code
        
        $form = $this->formElementManager->get(AlbumForm::class);
        
        // Other code
    }
}
```

Reply:
@froschdesign 
Very good, I thought you were going to skip the from element manager. 😄